lib/document/Document.js

```diff
@@ -41,6 +43,11 @@ class Document {
       delete data.$rev;
     }
 
+    if (Object.prototype.hasOwnProperty.call(rawDocument, '$meta')) {
+      this.meta = new Meta(rawDocument.$meta.userId);
```

Comment:
Let's pass the whole `rawDocument.$meta`. The document class should not know about meta internals.